components/url/lib.rs

```diff
@@ -163,6 +165,16 @@ impl ServoUrl {
 
 impl fmt::Display for ServoUrl {
     fn fmt(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        //changes for bug 22485
+        if self.0.to_string().as_bytes().len() > 40 {
```

Comment:
I think we can use `if self.0.len() > 40` here.

Reply:
this was a tricky one, self.0.len() can't be used it is wrapped in Arc as a thread
So i first need to convert it to string then loook for its length
which i did !

Reply:
Arc values can automatically be dereferenced to the wrapped value. Did you try it?

Reply:
yes it didn't worked but